### ISSUE TITLE
(1.19.2) internal exception    while looking at custom player heads

### ISSUE THREAD

**Issue by minecrafterseppe** [McJty/TheOneProbe#581]
internal exception: com.google.gson.JsonParseException: Don't know how to turn {} into a Component

mod version    1.19-6.2.1

(no crash)

**Comment by enigmaquip** [comment]
This is when looking at a player skull used as decoration added by When Dungeon's Arise
Example Skull Data:
`-10145, 159, 746 has the following block data: {z: 746, x: -10145, ForgeCaps: {}, id: "minecraft:skull", y: 159, SkullOwner: {Id: [I; -784904075, -953728851, -1127317950, 1997119024], Properties: {textures: [{Value: "eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvYTZlZjFjMjVmNTE2ZjJlN2Q2Zjc2Njc0MjBlMzNhZGNmM2NkZjkzOGNiMzdmOWE0MWE4YjM1ODY5ZjU2OWIifX19"}]}}}`

```
[10:20:04] [Netty Server IO #1/ERROR]: Received class net.minecraft.network.protocol.game.ClientboundDisconnectPacket that couldn't be processed
java.lang.ClassCastException: class net.minecraft.server.network.ServerGamePacketListenerImpl cannot be cast to class net.minecraft.network.protocol.game.ClientGamePacketListener (net.minecraft.server.network.ServerGamePacketListenerImpl and net.minecraft.network.protocol.game.ClientGamePacketListener are in module minecraft@1.19.2 of loader 'TRANSFORMER' @6321a5aa)
	at net.minecraft.network.protocol.game.ClientboundDisconnectPacket.m_5797_(ClientboundDisconnectPacket.java:7) ~[client-1.19.2-20220805.130853-srg.jar%23321!/:?]
	at net.minecraft.network.Connection.m_129517_(Connection.java:163) ~[client-1.19.2-20220805.130853-srg.jar%23321!/:?]
	at net.minecraft.network.Connection.channelRead0(Connection.java:148) [client-1.19.2-20220805.130853-srg.jar%23321!/:?]
	at net.minecraft.network.Connection.channelRead0(Connection.java:48) [client-1.19.2-20220805.130853-srg.jar%23321!/:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.local.LocalChannel.readInbound(LocalChannel.java:299) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.local.LocalChannel.finishPeerRead0(LocalChannel.java:445) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.local.LocalChannel.access$400(LocalChannel.java:50) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.channel.local.LocalChannel$5.run(LocalChannel.java:403) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) [netty-common-4.1.77.Final.jar%23106!/:4.1.77.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) [netty-common-4.1.77.Final.jar%23106!/:4.1.77.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) [netty-common-4.1.77.Final.jar%23106!/:4.1.77.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503) [netty-transport-4.1.77.Final.jar%23111!/:4.1.77.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995) [netty-common-4.1.77.Final.jar%23106!/:4.1.77.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.77.Final.jar%23106!/:4.1.77.Final]
	at java.lang.Thread.run(Thread.java:833) [?:?]
[10:20:04] [Server thread/INFO]: EnigmaQuip lost connection: Server sent an invalid packet
[10:20:04] [Server thread/INFO]: EnigmaQuip left the game
[10:20:04] [Server thread/INFO]: Stopping singleplayer server as player logged out```

**Comment by McJty** [comment]
I still need logs

**Comment by Harbivore** [comment]
I have the same issue. It isn't crashing minecraft, it kicks you out of the server (I'm on singleplayer) and then brings me to the multiplayer lobby screen. Happens when looking at custom player heads.

**Comment by OrangePieTaster** [comment]
The same issue happens to me when looking at a head, but not if it's on screen

**Comment by McJty** [comment]
I'll check it out soon.

**Comment by McJty** [comment]
I cannot reproduce this with vanilla player heads. Works fine for those. Perhaps it's an issue with the other mod?

**Comment by Harbivore** [comment]
Seems to be a problem with your mod but maybe only for these heads

Here is the issue on their modpage:
https://www.curseforge.com/minecraft/mc-mods/when-dungeons-arise/issues/109

Someone here fixed it by disabling The One Probe and changing it for Jade so it appears to be this mod.

**Comment by McJty** [comment]
It's not because it crashes in top that it is the fault of top. In most cases it isn't

**Comment by enigmaquip** [comment]
Looking at these skulls does return a GameProfile, but it has an empty name field
`com.mojang.authlib.GameProfile@529b13f4[id=296a6aa6-ef9b-467b-ae3e-31b8174a434c,name=<null>,properties={textures=[com.mojang.authlib.properties.Property@12834f5f]},legacy=false]`
Could you not also check for `profile.isComplete()` here https://github.com/McJtyMods/TheOneProbe/blob/1.19/src/main/java/mcjty/theoneprobe/apiimpl/providers/DefaultProbeInfoProvider.java#L163

**Comment by McJty** [comment]
Let me try that

**Comment by enigmaquip** [comment]
That worked in my testing with a personal compile. GameProfile does state that only one or the other of id or name have to defined, and isComplete just checks if both are indeed defined which works for making sure a name string is in fact defined

**Comment by McJty** [comment]
If Curseforge would cooperate I would release a fixed version

**Comment by Harbivore** [comment]
Thanks for fixing it so quick :)

**Comment by McJty** [comment]
Ok, should be fixed for 1.18 and 1.19 now

**Comment by McJty** [comment]
Because of the current CF problems the new releases are not yet approved. So you'll have to wait a bit
